src/DynamoDbCache.php

```diff
@@ -552,8 +558,13 @@ private function iterableToArray(iterable $iterable): array
 
     private function getKey(string $key): string
     {
-        if ($this->prefix !== null) {
-            return $this->prefix . $key;
+        if ($this->prefix) {
+            $key = $this->prefix . $key;
+        }
+
+        if (strlen($key) > self::MAX_KEY_LENGTH) {
+            $suffix = '_trunc_' . md5($key);
+            $key = substr($key, 0, self::MAX_KEY_LENGTH - strlen($suffix)) . $suffix;
```

Comment:
@RikudouSage I would solve it this way.

Reply:
I updated my PR to include your proposed changes.